apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1645,7 +1681,18 @@ func (u *UniterAPI) NetworkInfo(args params.NetworkInfoParams) (params.NetworkIn
 		}
 	}
 
-	networkInfos := machine.GetNetworkInfoForSpaces(spaces)
+	// Don't get the network info for the default space if there was an error
```

Comment:
The logic is there to maintain existing behaviour - I was careful to retain the semantics of the GetNetworkInfoForSpaces method. The only difference is that we pass in the address for the default space instead of having the method get it via machine.PrivateAddress().